[josejhkim] iP

josejhkim opened the pull request:
### John, your personalized chatbot 🤖

> Written goals have a way of transforming wishes into wants

Meet John, a personalized chatbot that stores your tasks, [based on the Duke chatbot](https://nus-cs2103-ay2021s1.github.io/website/se-book-adapted/projectDuke/index.html)

John can 
* Store your tasks **to a file**
* Mark them as done
* and more!

All you need to do is
1. Download it
2. Add your tasks
3. Mark them off as you get closer to your goals!

Features:

- [x] Managing tasks
- [ ] Sleek UI
- [ ] Task reminders

Everything is done through the `main` method inside `John.java`

Here's a little peek into how John works

```java
public void addTask(Task task) {
    this.taskList.add(task);
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

josejhkim committed:
change chatbot name to John and refactor files

josejhkim committed:
edit README to include new chatbot name

josejhkim committed:
change greeting message with new chatbot name

josejhkim committed:
import scanner for user input

josejhkim committed:
add echo user input functionality

josejhkim committed:
add string array to store user inputs

josejhkim committed:
replace string array with arraylist

josejhkim committed:
add feature for adding and printing list items

josejhkim committed:
create task class

josejhkim committed:
add method to task for getting description

josejhkim committed:
move Task class definition into separate file

josejhkim committed:
change taskList to take in Task instead of String

josejhkim committed:
change list function to use for loops for better indexing

josejhkim committed:
add functionality for marking/unmarking tasks

josejhkim committed:
rename string method to toString

josejhkim committed:
use toString() for printing out list

josejhkim committed:
create classes for different task types

josejhkim committed:
add support for different task types

josejhkim committed:
add output for invalid command

josejhkim committed:
replace '_' with '-'

josejhkim committed:
use while true loop instead

josejhkim committed:
add a line after goodbye response

josejhkim committed:
make John echo tasks that have been added

josejhkim committed:
add basic test case for i/o testing

josejhkim committed:
add try/catch for marking incorrect indices

josejhkim committed:
add error handling for incorrect command formats

josejhkim committed:
create basic exception files for John

josejhkim committed:
add basic error handling for wrong commands

josejhkim committed:
Merge branch 'master' of https://github.com/josejhkim/ip

josejhkim committed:
add basic delete command

josejhkim committed:
Create function for task creation message

josejhkim committed:
Define helper functions for creating tasks

josejhkim committed:
Make taskList into a private class field

josejhkim committed:
Define methods for marking tasks

josejhkim committed:
Change main() to use new helper and class methods

josejhkim committed:
Define method for deleting task

josejhkim committed:
Define method for printing task list

josejhkim committed:
Define method for writing task list to file

josejhkim committed:
Write sample tasks to data/john.txt

josejhkim committed:
Add files for testing file writing

josejhkim committed:
Merge branch 'branch-Level-7'

josejhkim committed:
Replace deadline date with LocalDate

josejhkim committed:
Parse user input for deadline as date

josejhkim committed:
Print deadline date in day-month-year format

josejhkim committed:
Merge branch 'branch-Level-8'

josejhkim committed:
Fix DateTimeFormatter to use dd instead of d

josejhkim committed:
Create class for managing the task list

josejhkim committed:
Create classes for parsing task strings

josejhkim committed:
Create class for dealing with the task list file

josejhkim committed:
Create class for user interaction messages

josejhkim committed:
Add method for printing task list

josejhkim committed:
Add method for checking if task list is empty

josejhkim committed:
Add method for retrieving read-only task list

josejhkim committed:
Fix incorrect method call for reading tasks from file

josejhkim committed:
Add methods for printing task addition/deletion

josejhkim committed:
Remove method for printing task addition

josejhkim committed:
Make task deletion return the deleted task

josejhkim committed:
Add method for retrieving task description

josejhkim committed:
Move task marking message to the Ui class

josejhkim committed:
Add method for printing notification for empty list

josejhkim committed:
Add method for printing notification for invalid command

josejhkim committed:
Add method for printing notification for file reading error

josejhkim committed:
Refactor John to use newly created classes

josejhkim committed:
Merge branch 'branch-MoreOOP'

josejhkim committed:
Refactor classes into respective packages

josejhkim committed:
Merge branch 'branch-Packages'

josejhkim committed:
Merge branch 'A-Gradle'

josejhkim committed:
Set up gradle for running John

josejhkim committed:
Add try/catch blocks for getDescription() indexing

josejhkim committed:
Add try/catch blocks for todo description parsing

josejhkim committed:
Parse deadline first before description

josejhkim committed:
Remove internal exception catching in createTask()

josejhkim committed:
Add string exception handling for readTodo()

josejhkim committed:
Write tests for markAsDone()

josejhkim committed:
Write tests for createTask()

josejhkim committed:
Configure gradle for building john.jar

josejhkim committed:
Make getTaskListFromFile() return a TaskList object

josejhkim committed:
Make taskList directly accept a TaskList from Storage

josejhkim committed:
Add JavaDoc headers for all methods

josejhkim committed:
Merge branch 'branch-A-JavaDoc'

josejhkim committed:
Apply coding standard to test files

josejhkim committed:
Add class headers for all classes

josejhkim committed:
Merge branch 'branch-A-JavaDoc'

josejhkim committed:
Apply coding standard to all files in main

josejhkim committed:
Apply coding standard to files in main and test

josejhkim committed:
Apply missed coding standards to all files

josejhkim committed:
Define method for returning a filtered task list

benedictleejr commented on src/main/java/john/Ui.java:
```diff
@@ -0,0 +1,54 @@
+        linebreak();
+    }
+
+    public void linebreak() {
```
# Naming for linebreak() method
To be consistent with the naming conventions, this method can be renamed to printLinebreak() 🦖

benedictleejr commented on src/main/java/john/John.java:
```diff
@@ -0,0 +1,78 @@
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class John {
```
# JavaDoc Comments
It'd be great to add JavaDoc comments on your public classes and methods, to help with code comprehension 👨‍💻

benedictleejr reviewed (commented):
Overall the code is clean, only issue is missing **JavaDoc comments** for your public classes and methods 🐘

bryonk commented on src/main/java/john/exception/JohnCommandException.java:
```diff
@@ -0,0 +1,4 @@
+package john.exception;
+
+public class JohnCommandException extends JohnException{
```
Just a question, is JohnCommandException being used anywhere else? If not maybe it could be removed if unnecessary.

bryonk commented on src/main/java/john/exception/JohnTaskException.java:
```diff
@@ -0,0 +1,4 @@
+package john.exception;
+
+public class JohnTaskException extends JohnException{
```
Same question as JohnCommandException, if the class is necessary.

bryonk reviewed (commented):
LGTM, overall good code quality, aside from the ones @benedictleejr mentioned. Other than that just 2 questions, regarding the exception classes.

josejhkim committed:
Add command for keyword searches

josejhkim committed:
Add feature for finding keyword in task list

josejhkim committed:
Add JavaDoc header for getFilteredTaskList()

josejhkim committed:
Rename linebreak() method to be a verb

josejhkim committed:
Define methods for getting messages as Strings

josejhkim committed:
Define methods for getting string version of task list

josejhkim committed:
Close the scanner after reading from file

josejhkim committed:
Define method for getting response as string

josejhkim committed:
Define method for getting size of task list

josejhkim committed:
Move error handling to John to get error messages as Strings

josejhkim committed:
Define methods for getting exception messages as Strings

josejhkim committed:
Save task list to file on by command

josejhkim committed:
Update gradle file for JavaFX

josejhkim committed:
Add basic GUI for John based on SE-Edu

josejhkim committed:
Merge branch 'Level-10'

josejhkim committed:
Delete unused exception classes

There are exception classes that extend from JohnException
which were originally intended to specify which exception occured
(e.g. invalid command, invalid task formatting).

However, all of the exceptions thrown by John and related methods
are simply JohnException, and the error messages themselves specify
what exception occured and why.

Thus to reduce project size and code complexity, these exceptions are being removed.

These exceptions can be added later if deemed necessary.

josejhkim committed:
Add assertions to methods

josejhkim committed:
Change deep nesting to use if-return instead

josejhkim committed:
Remove unused methods

josejhkim committed:
Make code quality changes

josejhkim committed:
Make code quality changes

josejhkim committed:
Make code quality changes

josejhkim committed:
Make code quality changes

josejhkim committed:
Improve code quality by giving meaning to magic numbers

josejhkim committed:
Improve code quality by adding line breaks for structure

josejhkim committed:
Improve code quality by breaking up long functions

josejhkim committed:
Remove unused scanner import

josejhkim committed:
Merge pull request #2 from josejhkim/branch-A-Assertions

Add assertions to methods

josejhkim committed:
Merge branch 'master' of https://github.com/josejhkim/ip

josejhkim committed:
Merge branch master

josejhkim committed:
Merge pull request #3 from josejhkim/branch-A-CodeQuality

josejhkim committed:
Add extension of tracking expenses

josejhkim committed:
Update gitignore to contain config and data directories

josejhkim committed:
Create method to get string with given exception

josejhkim committed:
Add "$" for printing total expense

josejhkim committed:
Fix minor bugs with the expense extension

josejhkim committed:
Make John print out specific error messages for tasks

josejhkim committed:
Create method for giving help on available commands

josejhkim committed:
Make John say hello on startup

josejhkim committed:
Add screenshot of the UI to docs

josejhkim committed:
Make John save the task list after every user input

josejhkim committed:
Update user guide

josejhkim committed:
Make all responses start with an upper case

josejhkim committed:
Update test cases for InputParser tests

josejhkim committed:
Make every exception thrown by TaskList a JohnException

josejhkim committed:
Fix error in exception message

josejhkim committed:
Fix typo in exception message

josejhkim committed:
Update and unify exception messages for wrong indexing

josejhkim committed:
Create method for getting error message with current list size

josejhkim committed:
Update invalid index error message to include lower bound

josejhkim committed:
Define a getter for the boolean value 'isDone'

josejhkim committed:
Remove unused methods

josejhkim committed:
Make expense double instead of an int

josejhkim committed:
Update test cases for TaskList

josejhkim committed:
Update JavaDocs to follow conventions

josejhkim committed:
Edit exception messages

josejhkim committed:
Handle case where there isn't a file to read from

josejhkim committed:
Edit test cases to reflect new error messages

josejhkim committed:
Edit code structure for checkstyle tests

josejhkim committed:
Fix formatting error with the user guide

josejhkim committed:
Fix formatting error with saving and printing expense

josejhkim committed:
Edit window size for MainWindow

josejhkim committed:
Change event to use LocalDate instead of string for dates

josejhkim committed:
Remove calculations outside of function parentheses

josejhkim committed:
Fix finding functionality showing original list

josejhkim committed:
Update UI example picture

josejhkim committed:
Update code structure for checkstyle tests

josejhkim committed:
Fix typo in task deletion message

josejhkim committed:
Remove deprecated storage file

josejhkim committed:
Reduce method size to make everything within 30 lines
